components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/core/ApplicationAuthenticatorManager.java

```diff
@@ -0,0 +1,175 @@
+import org.wso2.carbon.identity.application.common.ApplicationAuthenticatorService;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.UserDefinedFederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.UserDefinedLocalAuthenticatorConfig;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is used to manage the ApplicationAuthenticator instances.
+ */
+public class ApplicationAuthenticatorManager {
+
+    private static final ApplicationAuthenticatorManager instance = new ApplicationAuthenticatorManager();
+    private final List<ApplicationAuthenticator> systemDefinedAuthenticators = new ArrayList<>();
+
+    public static ApplicationAuthenticatorManager getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Add a new system authenticator.
+     *
+     * @param authenticator Authenticator to be added.
+     */
+    public void addSystemDefinedAuthenticator(ApplicationAuthenticator authenticator) {
+
+        systemDefinedAuthenticators.add(authenticator);
+    }
+
+    /**
+     * Remove a system authenticator.
+     *
+     * @param authenticator Authenticator to be removed.
+     */
+    public void removeSystemDefinedAuthenticator(ApplicationAuthenticator authenticator) {
+
+        systemDefinedAuthenticators.remove(authenticator);
+    }
+
+    /**
+     * Get all the system defined authenticators.
+     *
+     * @return List of system defined authenticators.
+     */
+    public List<ApplicationAuthenticator> getSystemDefinedAuthenticators() {
+
+        return systemDefinedAuthenticators;
+    }
+
+    /**
+     * Get the ApplicationAuthenticator for the given system defined authenticator configuration name.
+     *
+     * @param authenticatorName Authenticator name.
+     * @return  ApplicationAuthenticator instance.
+     */
+    public ApplicationAuthenticator getSystemDefinedAuthenticatorByName(String authenticatorName) {
+
+        for (ApplicationAuthenticator authenticator : systemDefinedAuthenticators) {
+            if (authenticator.getName().equals(authenticatorName)) {
+                return authenticator;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get all the authenticators for the given tenant domain.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return List of authenticators.
+     */
+    public List<ApplicationAuthenticator> getAllAuthenticators(String tenantDomain) {
+
+        List<ApplicationAuthenticator> allAuthenticators = new ArrayList<>(systemDefinedAuthenticators);
+
+        if (!ActionExecutorConfig.getInstance().isExecutionForActionTypeEnabled(ActionType.AUTHENTICATION) ||
+                FrameworkServiceDataHolder.getInstance().getUserDefinedAuthenticatorService() == null) {
+            return allAuthenticators;
+        }
```

Comment:
Well ActionExecutorConfig is not supposed to be used outside the component. It's within a private package. You should be able to read it from IdentityConfigParser.getInstance().getConfiguration().get(propertyKey)